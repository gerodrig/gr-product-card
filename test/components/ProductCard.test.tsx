import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components'
import { product2 } from '../data/products';

const { act } = renderer;


describe('tests in ProductCard component', () => {
    
    test('should render ProductCard component with the custom title', () => {
        const wrapper = renderer.create(
        <ProductCard product={product2} >
            {
                () => (
                    <h1>ProductCard</h1>
                )
            }
        </ProductCard>);

        expect(wrapper.toJSON()).toMatchSnapshot();
    }
    );

    test('should increment the counter', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2} >
                {
                    ({count, increaseBy}) => (
                        <>
                        <h1>ProductCard</h1>
                        <span>{ count }</span>
                        <button onClick={()=> increaseBy(1)}>increase</button>
                        </>
                    )
                }
            </ProductCard>);

            let tree = wrapper.toJSON();
            expect(tree).toMatchSnapshot();

            act(() => {
                // increase the counter
                (tree as any).children[2].props.onClick();
            });

            tree = wrapper.toJSON();

            expect((tree as any).children[1].children[0]).toBe('1')
    })
    

});

