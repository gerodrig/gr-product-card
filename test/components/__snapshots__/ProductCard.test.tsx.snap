// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in ProductCard component should increment the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    increase
  </button>
</div>
`;

exports[`tests in ProductCard component should render ProductCard component with the custom title 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
</div>
`;
